libcxx/test/std/numerics/c.math/isfinite.pass.cpp

```diff
@@ -62,9 +62,30 @@ struct TestInt {
   }
 };
 
+struct ConvertibleFloat {
+  int value;
+  ConvertibleFloat(int v) : value(v) {}
+  operator float() const { return static_cast<float>(value); }
+};
+
+struct ConvertibleDouble {
+  int value;
+  ConvertibleDouble(int v) : value(v) {}
+  operator double() const { return static_cast<double>(value); }
+};
+
+struct ConvertibleLongDouble {
+  int value;
+  ConvertibleLongDouble(int v) : value(v) {}
+  operator long double() const { return static_cast<long double>(value); }
+};
+
```

Comment:
```suggestion
template <class T>
struct ConvertibleTo {
  operator T() const { return {}; }
};
```
And then use `ConvertibleTo<float>` below.